core/src/test/java/org/apache/accumulo/core/file/rfile/bcfile/CompressionTest.java

```diff
@@ -18,109 +18,233 @@
-          Assert.assertNotNull(Compression.Algorithm.GZ.getCodec());
-          return true;
+        ArrayList<Future<Boolean>> results = Lists.newArrayList();
+
+        for (int i = 0; i < 30; i++) {
+          results.add(service.submit(new Callable<Boolean>()
+
+          {
+
+            @Override
+            public Boolean call() throws Exception {
+              Assert.assertNotNull(al + " should not be null", al.getCodec());
+              return true;
+            }
+
+          }));
         }
 
-      });
-    }
+        service.shutdown();
 
-    service.shutdown();
+        Assert.assertNotNull(al + " should not be null", codec);
 
-    Assert.assertNotNull(codec);
+        while (!service.awaitTermination(1, TimeUnit.SECONDS)) {
+          // wait
+        }
+
+        for (Future<Boolean> result : results) {
+          Assert.assertTrue(al + " resulted in a failed call to getcodec within the thread pool", result.get());
+        }
+      }
+    }
 
   }
 
   // don't start until we have created the codec
-  @Test
-  public void testManyDontStartUntilThread() throws IOException {
+  @Test(timeout = 60 * 1000)
+  public void testManyDontStartUntilThread() throws IOException, InterruptedException, ExecutionException {
+
+    for (final Algorithm al : Algorithm.values()) {
+      if (isSupported.get(al) != null && isSupported.get(al) == true) {
+
+        // first call to issupported should be true
+        Assert.assertTrue(al + " is not supported, but should be", al.isSupported());
 
-    ExecutorService service = Executors.newFixedThreadPool(10);
+        ExecutorService service = Executors.newFixedThreadPool(10);
 
-    for (int i = 0; i < 30; i++) {
+        ArrayList<Future<Boolean>> results = Lists.newArrayList();
 
-      service.submit(new Callable<Boolean>() {
+        for (int i = 0; i < 30; i++) {
 
-        @Override
-        public Boolean call() throws Exception {
-          Assert.assertNotNull(Compression.Algorithm.GZ.getCodec());
-          return true;
+          results.add(service.submit(new Callable<Boolean>() {
+
+            @Override
+            public Boolean call() throws Exception {
+              Assert.assertNotNull(al + " should have a non-null codec", al.getCodec());
+              return true;
+            }
+
+          }));
         }
 
-      });
-    }
+        service.shutdown();
 
-    service.shutdown();
+        while (!service.awaitTermination(1, TimeUnit.SECONDS)) {
+          // wait
+        }
+
+        for (Future<Boolean> result : results) {
+          Assert.assertTrue(al + " resulted in a failed call to getcodec within the thread pool", result.get());
+        }
+      }
+    }
 
   }
 
-  // don't start until we have created the codec
-  @Test
-  public void testThereCanBeOnlyOne() throws IOException, InterruptedException {
+  @Test(timeout = 60 * 1000)
+  public void testThereCanBeOnlyOne() throws IOException, InterruptedException, ExecutionException {
```

Comment:
Would an additional test like the following make sense?
* before threads call `createNewCodec(42)`  to create and cache a codec with that buffer size.. verify it differs from default
* have a bunch of threads call  `createNewCodec(42)` ... verify the get same codec